### PR TITLE
regeneration of thrift files. addition of glide for dependency manage…

### PR DESCRIPTION
**Pull request by sheenobu**
…ment

### CODE REVIEW

File: glide.lock [hunk 1]
```diff
@@ -0,0 +1,60 @@
+hash: 24dd85605b7c94a4ecc2c7fd16496787124c613e5fe272ec102575af738b01c0
```
Comment (chantra) [inline glide.lock:1]: i dont htink this file should be version controlled. Can you add a globbing exclusion of .lock file in .gitignore pls

File: src/shaping/shaping_linux.go [hunk 2]
```diff
@@ -168,7 +167,7 @@ func shape_on(id int64, shaping *atc_thrift.LinkShaping, link netlink.Link) erro
 	// rate was not set and is considered unlimited.
 	// In that case, let set the rate as high as we can.
 	// Note: currently, it is implemented as a u32 by the netlink library.
-	rate := uint64(shaping.GetRate() * 1000)
+	rate := uint64(shaping.Rate * 1000)
```
Comment (chantra) [inline src/shaping/shaping_linux.go:170]: Is this a change in thrift interface?

File: glide.lock (hunk 1 above)
Comment (sheenobu) [reply glide.lock:1]: We want to keep the lockfile so there is no surprise when building/testing. Different peoples copies of atc will be slightly different depending on when the project was downloaded and glide install ran.

This has been discussed pretty heavily in the ruby and rails community regarding Gemfile.lock.

File: src/shaping/shaping_linux.go (hunk 2 above)
Comment (sheenobu) [reply src/shaping/shaping_linux.go:170]: It's a change in how thrift generates golang code. GetProperty() is way less idiomatic than .Property. I am using an older version of the thrift-compiler (0.9.3 vs 0.9.1) but the thrift library that provides the golang support and generation code is much newer.

**Pull request merged**